plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/TransformPom.java

```diff
@@ -34,7 +35,7 @@ public boolean check(Map<String, Object> info) {
         boolean parentV2 = parent.compareVersionTo("2.0") >= 0;
         boolean coreRequiresNewParentPOM = coreCoordinates.compareVersionTo(CORE_NEW_PARENT_POM) >= 0;
 
-        if (isBO || isCB || (pluginPOM && parentV2)) {
+        if (isDeclarativePipeline || isBO || isCB || (pluginPOM && parentV2)) {
```

Comment:
We should look for a better way to generalize this (possible in the hooks). Can be in a follow up.